core/ir/instr_shared.c

```diff
@@ -1105,7 +1105,8 @@ instr_set_label_callback(instr_t *instr, instr_label_callback_t cb)
 {
     CLIENT_ASSERT(instr_is_label(instr),
                   "only set callback functions for label instructions");
-    CLIENT_ASSERT(instr->label_cb == NULL, "label callback function is already set");
+    CLIENT_ASSERT(instr->label_cb == NULL || cb == NULL,
```

Comment:
I hit this when instr_clone was called during trace building (not sure how it was never hit before: I think b/c there aren't good tests of the emulation API).  instr_clone says:
```C
    } else if (instr_is_label(orig)) {
        /* We don't know what this callback does, we can't copy this. The caller that
         * makes the clone needs to take care of this, xref i#3926.
         */
        instr_set_label_callback(instr, NULL);
```
And the docs for instr_set_label_callback say:
```C
 * \note This data field is not copied across instr_clone(). Instead, the
 * clone's field will be NULL (xref i#3962).
```
So yes this is allowing removal of the callback for the clone, which was supposed to already work but did not.

I think this is required for any use of emulation labels so if a different PR it would have to go in first.  I could do that.

Reply:
PR #4889.

Reply:
LGTM, just reviewed.